src/Common/src/TypeSystem/Interop/IL/Marshaller.cs

```diff
@@ -243,13 +375,8 @@ public bool IsMarshallingRequired()
         private TypeDesc GetNativeTypeFromMarshallerKind(TypeDesc type, MarshallerKind kind, MarshallerKind elementMarshallerKind,
                 MarshalAsDescriptor marshalAs)
         {
-            if (type.IsByRef)
-            {
-                return GetNativeTypeFromMarshallerKind(((ByRefType)type).ParameterType, kind, elementMarshallerKind, marshalAs).MakePointerType();
-            }
-
             TypeSystemContext context = PInvokeMethodData.Context;
-            NativeType nativeType = NativeType.Invalid;
+            NativeType nativeType = TypeSystem.NativeType.Invalid;
```

Comment:
Nit: may be  define "using NaitveTypeKind = TypeSystem.NativeType;" at the top, so that you don't have to qualify NativeType everywhere and it doesn't clash with the property NativeType.